src/generic/PermGroups.jl

```diff
@@ -340,15 +340,34 @@ function setpermstyle(format::Symbol)
    return format
 end
 
-function show(io::IO, g::Perm)
+function Base.show(io::IO, g::Perm)
    if _permdisplaystyle.format == :array
       print(io, "[" * join(g.d, ", ") * "]")
    elseif _permdisplaystyle.format == :cycles
-      cd = cycles(g)
-      if g == parent(g)()
-         print(io, "()")
+      print(io, _permstring(g, last(displaysize(io))))
+   end
+end
+
+function _permstring(p::Perm, width::Integer)
```

Comment:
This method could probably be made eventually to accept an `io` argument and writing to it (via the `join(io, ...)` method), instead of creating a big string. But who cares, until we get complaints :)